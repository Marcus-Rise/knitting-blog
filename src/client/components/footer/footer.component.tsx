import type { FC } from "react";
import React from "react";
import styles from "./footer.module.scss";

interface IProps {
  year: string | number;
  author: string;
  authorLink: string;
}

const Footer: FC<IProps> = ({ author, authorLink, year }) => (
  <footer className={styles.footer}>
    <span className="mr-1">&#9400; {year}</span>
    <a href={authorLink} target={"_blank"} rel="noreferrer">
      {author}
    </a>
  </footer>
);

export { Footer };
