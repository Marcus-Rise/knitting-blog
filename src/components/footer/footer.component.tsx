import React from "react";
import styles from "./footer.module.scss";

interface IProps {
  year: string | number;
  author: string;
  authorLink: string;
}

const Footer: React.FC<IProps> = (props) => {
  return (
    <footer className={styles.footer}>
      <span className="mr-1">&#9400; {props.year}</span>
      <a href={props.authorLink} target={"_blank"} rel="noreferrer">
        {props.author}
      </a>
    </footer>
  );
};

export { Footer };
