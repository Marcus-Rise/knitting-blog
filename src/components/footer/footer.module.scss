.footer {
  background-color: #dbdbdb;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 1rem;
  gap: 1rem;
}

.link {
  text-decoration: none;
  color: inherit;
}
