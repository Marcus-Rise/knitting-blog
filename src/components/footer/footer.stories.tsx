import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { Footer } from "./footer.component";

const Config: ComponentMeta<typeof Footer> = {
  title: "components/Footer",
  component: Footer,
};

const Template: ComponentStory<typeof Footer> = (args) => <Footer {...args} />;

const Default = Template.bind({});
Default.args = {
  authorLink: "https://marcus-rise.dev",
  authorName: "MarcusRise",
  year: "2022",
};

export default Config;
export { Default };
