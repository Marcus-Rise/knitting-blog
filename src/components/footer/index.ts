"use client";

export * from "./footer.component";
