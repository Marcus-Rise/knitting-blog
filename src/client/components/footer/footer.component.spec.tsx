import { render } from "@testing-library/react";
import { Footer } from "./footer.component";
import React from "react";

describe("Footer", () => {
  test("render", () => {
    const { asFragment } = render(<Footer year={2020} author={"Author"} authorLink={"http"} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
