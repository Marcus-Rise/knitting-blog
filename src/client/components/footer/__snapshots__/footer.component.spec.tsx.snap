// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <span
      class="mr-1"
    >
      â’¸ 2020
    </span>
    <a
      href="http"
      rel="noreferrer"
      target="_blank"
    >
      Author
    </a>
  </footer>
</DocumentFragment>
`;
